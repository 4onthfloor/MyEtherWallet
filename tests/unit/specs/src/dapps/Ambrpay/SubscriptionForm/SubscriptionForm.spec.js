import SubscriptionForm from '@/dapps/Ambrpay/containers/SubscriptionForm/SubscriptionForm.vue';
import { shallowMount } from '@vue/test-utils';
import Blockie from '@/components/Blockie';
import Vue from 'vue';
import { Tooling } from '@@/helpers';

describe('SubscriptionForm.vue', () => {
  let localVue, wrapper, i18n, store;

  beforeAll(() => {
    document.execCommand = jest.fn().mockImplementation(command => {
      return command;
    });

    Vue.toasted = jest.fn().mockImplementation(global => {
      return global;
    });

    const baseSetup = Tooling.createLocalVueInstance();
    localVue = baseSetup.localVue;
    i18n = baseSetup.i18n;
    store = baseSetup.store;
  });

  beforeEach(() => {
    wrapper = shallowMount(SubscriptionForm, {
      localVue,
      i18n,
      store,
      attachToDocument: true,
      stubs: {
        blockie: Blockie
      }
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be a vue instance', () => {
    expect(wrapper.isVueInstance).toBeTruthy();
  });

  it('should return the correct data', () => {
    expect(wrapper.vm.$data.address).toEqual('');
    expect(wrapper.vm.$data.isValidAddress).toEqual(false);
    expect(wrapper.vm.$data.hexAddress).toEqual('');
    expect(wrapper.vm.$data.intervalDays).toEqual('');
    expect(wrapper.vm.$data.sendAmount).toEqual('');
    expect(wrapper.vm.$data.amountErrMsg).toEqual('');
    expect(wrapper.vm.$data.intervalErrMsg).toEqual('');
  });

  it('should set an amount error message if sendAmount < 0.01', async () => {
    const sendAmount = 0.001;
    wrapper.setData({ sendAmount });
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$data.amountErrMsg).toEqual(
      'The minimum amount is 0.01 or greater'
    );
  });

  it('should set an amount error message if sendAmount > account balance', async () => {
    const sendAmount = 1;
    wrapper.setData({ sendAmount });
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$data.amountErrMsg).toEqual(
      'Amount higher than balance (including 1% automation fee)'
    );
  });

  it('should set an interval error message if the number is invalid', async () => {
    const intervalDays = '000';
    wrapper.setData({ intervalDays });
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.$data.intervalErrMsg).toEqual(
      'Please enter a correct number'
    );
  });

  it('should call sendEntireBalance on click to entire balance', () => {
    wrapper.setData({ sendAmount: '3' });
    wrapper.find('.entire-balance').trigger('click');
    expect(wrapper.vm.$data.sendAmount).toBe('0');
  });

  it('should send $emit on button click', () => {
    wrapper.find('.mew-btn').trigger('click');
    wrapper.vm.$emit('startSubscription');

    expect(wrapper.emitted().startSubscription).toBeTruthy();
  });

  it('should clear the form', () => {
    wrapper.setData({
      address: '0xDECAF9CD2367cdbb726E904cD6397eDFcAe6068D',
      isValidAddress: true,
      hexAddress: '0xDECAF9CD2367cdbb726E904cD6397eDFcAe6068D',
      intervalDays: '5',
      sendAmount: '1',
      amountErrMsg: 'There is an error',
      intervalErrMsg: 'There is an error'
    });
    wrapper.find('.clear-all-btn').trigger('click');
    expect(wrapper.vm.$data.address).toEqual('');
    expect(wrapper.vm.$data.isValidAddress).toEqual(false);
    expect(wrapper.vm.$data.hexAddress).toEqual('');
    expect(wrapper.vm.$data.intervalDays).toEqual('');
    expect(wrapper.vm.$data.sendAmount).toEqual('');
    expect(wrapper.vm.$data.amountErrMsg).toEqual('');
    expect(wrapper.vm.$data.intervalErrMsg).toEqual('');
  });

  it('should dismount properly', () => {
    expect(wrapper.destroy()).toBe(undefined);
  });
});
