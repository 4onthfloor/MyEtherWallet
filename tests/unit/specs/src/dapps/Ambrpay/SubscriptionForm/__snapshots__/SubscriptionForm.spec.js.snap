// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionForm.vue should render correctly 1`] = `
<div class="subscription-container">
  <div class="subscription-content">
    <div class="subscription-form-container">
      <b-container-stub tag="div">
        <b-row-stub tag="div" class="address-block">
          <dropdown-address-selector-stub title="To Address" currency="ETH"></dropdown-address-selector-stub>
        </b-row-stub>
        <b-row-stub tag="div">
          <b-col-stub cols="12" md="5" tag="div" class="mt-3"><span class="label-text">Type</span>
            <div class="fake-input">
              <p><img src="@/assets/images/currency/eth.svg" class="currency-icon"> <span class="token-txt">ETH</span>-Ethereum
              </p>
            </div>
          </b-col-stub>
          <b-col-stub cols="12" md="7" tag="div" class="amount-container mt-3">
            <div class="amount-text"><span class="label-text">Amount</span> <span class="action-text entire-balance">Entire Balance</span></div> <input type="number" placeholder="0.01" min="0.01">
            <p class="sub-text">1% automation-fee will be added on top</p>
            <p class="sub-text err-msg" style="display: none;">

            </p>
          </b-col-stub>
        </b-row-stub>
        <b-row-stub tag="div" class="row-style mt-1 interval-container"><span class="label-text">Interval in Every</span> <span class="days-text" style="display: none;">Day(s)</span> <input placeholder="Enter the number of day(s)" type="number" min="1" max="365">
          <p class="sub-text err-msg" style="display: none;">

          </p>
        </b-row-stub>
        <b-row-stub tag="div" class="mt-5 button-container">
          <b-button-stub target="_self" event="click" routertag="a" variant="secondary" type="button" tag="button" class="active-sub-btn my-subscriptions-container">Manage Subscriptions
          </b-button-stub>
          <b-button-stub target="_self" event="click" routertag="a" variant="secondary" type="button" tag="button" class="mew-btn disabled">Start Recurring</b-button-stub>
        </b-row-stub>
        <b-row-stub tag="div">
          <div class="clear-all-btn mx-auto mt-3">
            Clear All
          </div>
        </b-row-stub>
      </b-container-stub>
    </div>
  </div>
</div>
`;
